analysis/diabetes_analysis.ipynb

```diff
@@ -0,0 +1,1602 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "56ae3098-c2d3-49f8-8019-b8ed4ddd78df",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import altair as alt\n",
+    "from sklearn.pipeline import Pipeline, make_pipeline\n",
+    "from sklearn.model_selection import (\n",
+    "    RandomizedSearchCV,\n",
+    "    cross_validate,\n",
+    "    train_test_split,\n",
+    ")\n",
+    "from sklearn.preprocessing import StandardScaler\n",
+    "from sklearn.pipeline import Pipeline, make_pipeline\n",
+    "from sklearn.compose import make_column_transformer\n",
+    "from sklearn.linear_model import LogisticRegression\n",
+    "from sklearn.metrics import (\n",
+    "    fbeta_score, \n",
+    "    confusion_matrix,\n",
+    "    ConfusionMatrixDisplay,\n",
+    "    PrecisionRecallDisplay,\n",
+    "    RocCurveDisplay\n",
```

Comment:
Maybe add a caption under each display figure (i.e. Fig 1. results from the confusion matrix shows... ) to help the reader understand what's happening in the displayed figure.